### PR TITLE
[hexl] Update version to 1.2.1

### PR DESCRIPTION
**Pull request by GelilaSeifu**
**Describe the pull request**

- #### What does your PR fix?  
  `Update hexl version to 1.2.1`
- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  `x64 & (windows | linux | osx) & static , No`

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  `Yes`

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/intel-hexl/vcpkg.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "hexl",
-  "version": "1.1.0",
+  "version": "1.2.0",
   "description": "Intel® HEXL is an open-source library which provides efficient implementations of integer arithmetic on Galois fields.",
-  "homepage": "https://github.com/intel/hexl",
+  "homepage": "https://gitlab.devtools.intel.com/",
```
Comment (fboemer) [inline ports/intel-hexl/vcpkg.json:5]:
```suggestion
  "homepage": "https://github.com/intel/hexl",
```

**Comment by JonLiu1993** [comment]
@GelilaSeifu ,Thanks for your pr,could you please take a look:
https://dev.azure.com/vcpkg/public/_build/results?buildId=57509&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by JonLiu1993** [comment]
This is the error log:
````
CMake Error at D:/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/share/cmake-3.20/Modules/ExternalProject.cmake:2633 (message):
  error: could not find git for clone of cpu_features
Call Stack (most recent call first):
  D:/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/share/cmake-3.20/Modules/ExternalProject.cmake:3681 (_ep_add_download_command)
  CMakeLists.txt:9 (ExternalProject_Add)


CMake Error at cmake/cpu-features/CMakeLists.txt:10 (message):
  CMake step for cpu-features failed: 1

````

**Comment by GelilaSeifu** [comment]
> This is the error log:
> 
> ```
> CMake Error at D:/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/share/cmake-3.20/Modules/ExternalProject.cmake:2633 (message):
>   error: could not find git for clone of cpu_features
> Call Stack (most recent call first):
>   D:/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/share/cmake-3.20/Modules/ExternalProject.cmake:3681 (_ep_add_download_command)
>   CMakeLists.txt:9 (ExternalProject_Add)
> 
> 
> CMake Error at cmake/cpu-features/CMakeLists.txt:10 (message):
>   CMake step for cpu-features failed: 1
> ```

**Comment by JonLiu1993** [comment]
same error:
````
CMake Error at D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:2650 (message):
  error: could not find git for clone of cpu_features
Call Stack (most recent call first):
  D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:3700 (_ep_add_download_command)
  CMakeLists.txt:9 (ExternalProject_Add)


CMake Error at cmake/cpu-features/CMakeLists.txt:10 (message):
  CMake step for cpu-features failed: 1
````

**Comment by GelilaSeifu** [comment]
> same error:
> 
> ```
> CMake Error at D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:2650 (message):
>   error: could not find git for clone of cpu_features
> Call Stack (most recent call first):
>   D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:3700 (_ep_add_download_command)
>   CMakeLists.txt:9 (ExternalProject_Add)
> 
> 
> CMake Error at cmake/cpu-features/CMakeLists.txt:10 (message):
>   CMake step for cpu-features failed: 1
> ```


@JonLiu1993 I am not able to reproduce this on my dev system. I am able to build hexl on Windows System with both triplets from failing jobs just by passing -DVCPKG_TARGET_TRIPLET=x64-windows-static or x64-windows-static-md. Is there something I am missing?

**Comment by JonLiu1993** [comment]
> > same error:
> > ```
> > CMake Error at D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:2650 (message):
> >   error: could not find git for clone of cpu_features
> > Call Stack (most recent call first):
> >   D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:3700 (_ep_add_download_command)
> >   CMakeLists.txt:9 (ExternalProject_Add)
> > 
> > 
> > CMake Error at cmake/cpu-features/CMakeLists.txt:10 (message):
> >   CMake step for cpu-features failed: 1
> > ```
> 
> @JonLiu1993 I am not able to reproduce this on my dev system. I am able to build hexl on Windows System with both triplets from failing jobs just by passing -DVCPKG_TARGET_TRIPLET=x64-windows-static or x64-windows-static-md. Is there something I am missing?

I cannot reproduce this error locally too, I will investigate this issue further

File: ports/hexl/portfile.cmake [hunk 2]
```diff
@@ -22,18 +21,13 @@ vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
     DISABLE_PARALLEL_CONFIGURE
     OPTIONS
-        "-DHEXL_DEBUG=OFF"
         "-DHEXL_BENCHMARK=OFF"
-        "-DHEXL_EXPORT=ON"
         "-DHEXL_COVERAGE=OFF"
         "-DHEXL_TESTING=OFF"
         "-DHEXL_SHARED_LIB=${HEXL_SHARED}"
```
Comment (JackBoosY) [inline ports/hexl/portfile.cmake:27]:
```suggestion
        "-DHEXL_SHARED_LIB=${HEXL_SHARED}"
        -DGIT_EXECUTABLE=${GIT}
```

**Comment by JackBoosY** [comment]
When building x64-windows-static:
```
G:\19468\vcpkg\buildtrees\hexl\src\ec9afec77e-a8284b241f\hexl\util/msvc.hpp(16): warning C4163: '_udiv128': not available as an intrinsic function
G:\19468\vcpkg\buildtrees\hexl\src\ec9afec77e-a8284b241f\hexl\util/msvc.hpp(29): error C3861: '_udiv128': identifier not found
G:\19468\vcpkg\buildtrees\hexl\src\ec9afec77e-a8284b241f\hexl\util/msvc.hpp(162): warning C4296: '>=': expression is always true
G:\19468\vcpkg\buildtrees\hexl\src\ec9afec77e-a8284b241f\hexl\util/msvc.hpp(168): warning C4296: '>=': expression is always true
```

**Comment by GelilaSeifu** [comment]
> When building x64-windows-static:
> 
> ```
> G:\19468\vcpkg\buildtrees\hexl\src\ec9afec77e-a8284b241f\hexl\util/msvc.hpp(16): warning C4163: '_udiv128': not available as an intrinsic function
> G:\19468\vcpkg\buildtrees\hexl\src\ec9afec77e-a8284b241f\hexl\util/msvc.hpp(29): error C3861: '_udiv128': identifier not found
> G:\19468\vcpkg\buildtrees\hexl\src\ec9afec77e-a8284b241f\hexl\util/msvc.hpp(162): warning C4296: '>=': expression is always true
> G:\19468\vcpkg\buildtrees\hexl\src\ec9afec77e-a8284b241f\hexl\util/msvc.hpp(168): warning C4296: '>=': expression is always true
> ```

At the moment, we don't support Ninja and we do use some intrinsic from MSVC.

**Comment by JackBoosY** [comment]
I will take a look.

**Comment by GelilaSeifu** [comment]
> I will take a look.

Is there a way one can specify the generator to be "Visual Studio 16 2019" instead of Ninja for the pipeline?

**Comment by JackBoosY** [comment]
@GelilaSeifu Add `WINDOWS_USE_MSBUILD` to `vcpkg_cmake_configure`.

**Comment by JackBoosY** [comment]
Please ping me if your PR is ready for review.

**Comment by JackBoosY** [comment]
The reason why it ran failed is:
CMakeLists.txt:
```
configure_file(CMakeLists.txt.in cpu-features-download/CMakeLists.txt)
execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
  RESULT_VARIABLE result
  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cpu-features-download )
```

CMakeLists.txt.in:
```
include(ExternalProject)
ExternalProject_Add(cpu_features
  GIT_REPOSITORY    https://github.com/google/cpu_features.git
  GIT_TAG           master
  SOURCE_DIR        "${CMAKE_CURRENT_BINARY_DIR}/cpu-features-src"
  BINARY_DIR        "${CMAKE_CURRENT_BINARY_DIR}/cpu-features-build"
  CONFIGURE_COMMAND ""
  BUILD_COMMAND     ""
  INSTALL_COMMAND   ""
  TEST_COMMAND      ""
  UPDATE_COMMAND    ""
)
```

And the reason why pass -DGIT_EXECUTABLE=${GIT}` doesn't work is we didn't pass `GIT_EXECUTABLE` to `execute_process`.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
> > I need to confirm something:
> > In https://github.com/intel/hexl/blob/main/hexl/CMakeLists.txt#L208-L213:
> > ```cmake
> >                 add_custom_command(TARGET hexl POST_BUILD
> >                     COMMAND lib.exe /OUT:$<TARGET_FILE:hexl>
> >                         $<TARGET_FILE:hexl>
> >                         $<TARGET_FILE:cpu_features>
> >                     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
> >                     DEPENDS hexl cpu_features
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > What's the purpose of this?
> > We have a dependency on cpu_features and if users choose not to install it from vcpkg we provide a way to build it from source and add the dependency to hexl.

So we can also use `target_link_libraries` to `cpu_features`, right?

Since we use vcpkg internal dependencies, we don't need to find git now.

**Comment by GelilaSeifu** [comment]
> /azp run

@BillyONeal  Do you mind rerunning the pipeline? I don't believe the failed jobs are related to the PR.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**